DB/src/main/java/io/deephaven/db/util/PythonScopeJpyImpl.java

```diff
@@ -67,6 +67,72 @@ public CallableWrapper(PyObject pyObject) {
         public Object call(Object... args) {
             return convert(pyObject.callMethod("__call__", args));
         }
+
+        public PyObject getPyObject() {
+            return pyObject;
+        }
+    }
+
+    public static final class NumbaCallableWrapper extends CallableWrapper {
+        private List<Class> paramTypes;
+        private Class returnType;
+
+        public NumbaCallableWrapper(PyObject pyObject, Class returnType, List<Class> paramTypes) {
+            super(pyObject);
+            this.returnType = returnType;
+            this.paramTypes = paramTypes;
+        }
+
+        public Class getReturnType() {
+            return returnType;
+        }
+
+        public List<Class> getParamTypes() {
+            return paramTypes;
+        }
+    }
+
+    private static CallableWrapper wrapCallable(PyObject pyObject) {
+        if (pyObject.hasAttribute("ufunc")) {
+            List<PyObject> params = pyObject.getAttribute("types").asList();
+            if (params.isEmpty()) {
+                return new CallableWrapper(pyObject);
+            }
```

Comment:
Is it possible to have a vectorized ufunc that doesn't take any arguments, but does have a vectorized return type? Does it make sense to have this case handled w/ NumbaCallableWrapper?

Edit: coming back here later, it seems like "types" is a ufunc specific datastructure, and we never expect it to be empty, right? In which case, I think we should throw an exception if we see this case.

Reply:
good suggestion based on 'fail early fail fast', added exceptions